packages/block-library/src/navigation-link/index.js

```diff
@@ -11,11 +11,12 @@ import { InnerBlocks } from '@wordpress/block-editor';
 import metadata from './block.json';
 import edit from './edit';
 import save from './save';
-import variations from './variations';
 
 const { name } = metadata;
 
-export { metadata, name };
+// If we export metadata, this calls unstable__bootstrapServerSideBlockDefinitions a second time, and overwrites
+// what was injected on the page
+export { name };
```

Comment:
So I noticed that we were calling `unstable__bootstrapServerSideBlockDefinitions` twice. Once when looking at the injected server definitions, and a second time during block register. The local definition would overwrite the server definition. I'm not sure if this is what y'all intended, since I also see this behavior for other blocks.

Reply:
It's a complex issue and not that simple to solve for all blocks. When you remove this client-side metadata, then it no longer is exposed to unit tests. In addition to that, some block definitions are shared with the mobile app or used with a standalone block editor (like in Storybook). We need to solve it on the framework level. In the context of WordPress, it could work exactly as you changed it.

Reply:
Oh interesting, it was intended to be overwritten by the client. As a placeholder I've added an extra check in unstable__bootstrapServerSideBlockDefinitions, but I'll be sure to highlight that.

Reply:
> Oh interesting, it was intended to be overwritten by the client.

I don't think it was. We had a very long process of adding server-side registration that is based on `block.json` and we never considered that it might create issues until recently as discussed 😄

Reply:
I opened https://github.com/WordPress/gutenberg/pull/29213 which contains only the change that prevents overriding metadata set on the server.

Reply:
This one should be solved now by https://github.com/WordPress/gutenberg/pull/29302 thanks to @gziolo